node/src/components/consensus/era_supervisor.rs

```diff
@@ -1053,7 +1046,7 @@ where
                         .collect(),
                 });
                 let finalized_block = FinalizedBlock::new(
-                    value,
+                    Arc::try_unwrap(value).unwrap_or_else(|arc| (*arc).clone()),
```

Comment:
Can you extract this to a method like discussed? Unless it's used only in this single place.

Reply:
It's only used here, but it would still be nicer.

Anyway, I was considering making a follow-up PR with cleanups anyway, and limiting this PR to just the backports.

Reply:
That's fine.